<launch>
  <arg name="UAV_NAME" default="$(optenv UAV_NAME uav1)"/>

  <group ns="$(arg UAV_NAME)">

    <!-- Voxblox ESDF -->
    <node pkg="voxblox_ros"
          type="esdf_server"
          name="esdf_server">
          <!-- output="screen"> -->

      <rosparam command="load"
                file="$(find cbf_filter)/config/esdf.yaml"/>
      <!-- LiDAR cloud -->
      <remap from="pointcloud"
             to="livox/points"/>
    </node>

    <!-- CBF filter (unchanged) -->
    <node pkg="cbf_filter"
          type="cbf_filter_node"
          name="cbf_filter_node"
          output="screen">
      <rosparam command="load"
                file="$(find cbf_filter)/config/cbf.yaml"/>

      <remap from="~input_reference"  to="control_manager/velocity_reference"/>
      <remap from="~odometry"         to="estimation_manager/odom_main"/>
      <remap from="~filtered_reference"           to="cbf_filter/filtered_reference"/>
      <remap from="~cbf_gradient_marker"          to="cbf_filter/gradient_marker"/>
      <remap from="~cbf_nearest_obstacle_marker"  to="cbf_filter/nearest_obstacle_marker"/>
    </node>

  </group>
</launch>
